src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -32,8 +32,14 @@ public class InsertOptionalDataVisitor : SarifRewritingVisitor
         private const string Email = nameof(Email);
         private const string CommitSha = nameof(CommitSha);
 
-        public InsertOptionalDataVisitor(OptionallyEmittedData dataToInsert, Run run, IEnumerable<string> insertProperties)
-            : this(dataToInsert, run?.OriginalUriBaseIds, insertProperties: insertProperties)
+        public InsertOptionalDataVisitor(OptionallyEmittedData dataToInsert,
```

Comment:
>InsertOptionalDataVisitor [](http://example.com/codeflow?start=15&length=25)

This helper did not previously allow for specifying the file regions cache. Our overloads in this type are unfortunate.

The problem is specifying original uri base ids independently vs. providing a run instance...